### PR TITLE
Add ProtocolTracers parameter to Ouroboros.Consensus.Node.run

### PR DESCRIPTION
**Pull request by intricate**
Requirement for https://github.com/input-output-hk/cardano-node/issues/527.

Specifically:

>We also need to trace at the default log level for the outbound side of tx submission:
>* when a remote peer has requested a tx id, and we do send the tx to them, then we should log that the txid was requested and sent

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Node.hs [hunk 1]
```diff
@@ -146,7 +148,7 @@ run tracers chainDbTracer diffusionTracers diffusionArguments networkMagic
                            ()
           networkApps = consensusNetworkApps
             nodeKernel
-            nullProtocolTracers
+            protocolTracers
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: Huh, I'm surprised to see this. I remember seeing flags in the node to trace the different protocols, apparently they didn't work?

Comment (intricate) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: Yeah so the protocol tracers were never being passed through.

**Comment by mrBliss** [comment]
What's left to do?

**Comment by intricate** [comment]
> What's left to do?

Nothing, this is good to go. Just marked the PR as "ready for review".

File: ouroboros-consensus/src/Ouroboros/Consensus/Node.hs [hunk 2]
```diff
@@ -89,12 +90,13 @@ data IsProducer
 run
   :: forall blk.
      RunNode blk
-  => Tracers IO ConnectionId blk          -- ^ Consensus tracers
-  -> Tracer  IO (ChainDB.TraceEvent blk)  -- ^ ChainDB tracer
-  -> DiffusionTracers                     -- ^ Diffusion tracers
-  -> DiffusionArguments                   -- ^ Diffusion arguments
+  => Tracers IO ConnectionId blk                             -- ^ Consensus tracers
+  -> ProtocolTracers IO ConnectionId blk DeserialiseFailure  -- ^ Protocol tracers
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: Nitpick: this line is now longer 80 chars. Suggestion:
```haskell
  => Tracers IO ConnectionId blk          -- ^ Consensus tracers
  -> ProtocolTracers IO ConnectionId blk DeserialiseFailure
     -- ^ Protocol tracers
  -> Tracer  IO (ChainDB.TraceEvent blk)  -- ^ ChainDB tracer
```
You can revert the changes that realign the Haddock comments.
